### DISCUSSION TITLE
Pico with PIO is a beast

### DISCUSSION THREAD

**Discussion by joey120373** [Bodmer/TFT_eSPI#1623] [General]
I’m developing code for a personal project, using the ILI9341 based TFT because the hardware scroll feature makes it super easy to implement a “serial terminal” . That is one part of my project that is crucial. If anyone knows an easy way to get a smooth scrolling terminal type window using larger screens ( ILI9488, SSD1963 etc. ) please let me know! 
Back to the point, depending on the display manufacturer, I can get somewhere in the neighborhood of 1000+fps with the sprite rotating cube demo, and ~350fps with the boing_ball demo. The UTFT demo is done in 
145 millisecs. 
This is all in 8 bit parallel mode and the RP2040 clocked at 250Mhz, and optimizes to “fast”

Using very short connection wires, and not all of my displays are happy at 250Mhz, most of them seem to be stable at 150-200Mhz. But even at those slower clocks the performance is quite impressive. 

Thanks for all the hard work on this library! 

Is there any chance we might see 16 bit parallel support for the RP2040 and the SSD1963? 

With that many pixels to fill (480x800) the little RP2040 does pretty good, but 16 bit writes would probably make it a bit more snappy for user interfaces .

**Comment by Bodmer** [comment]
Yes, it is very fast. The RP2040 PIO is ideal for this application.

16 bit is planned and should be easy to implement now I have understood how to program the PIO.

**Comment by Bodmer** [comment]
Many screens support hardware scrolling, the command byte needed might be different.
